src/operators/flat_map.rs

```diff
@@ -26,17 +26,19 @@ where
 
 impl<In, Out, F, I> Operator<In, Out> for FlatMap<In, Out, F, I>
 where
-    In: Clone,
-    F: Fn(In) -> I,
+    F: Fn(&In) -> I,
     I: IntoIterator<Item = Out>,
 {
     fn process_changes(&mut self, input_changes: &[Update<In>]) -> Vec<Update<Out>> {
         input_changes
             .iter()
             .flat_map(|change| {
-                (self.f)(change.item().clone())
-                    .into_iter()
-                    .map(move |item| change.as_ref().map(|_| item))
+                (self.f)(&change.item).into_iter().map(move |item| {
```

Comment:
```suggestion
                (self.f)(change.item).into_iter().map(|item| {
```

Try from here. No need for a reference, it can be my ved into the function. The `move` on the closure is tripping you up.

Reply:
Where did this change go? You almost had it right but keep going backwards. You don't need references or clone for this implementation. Trust me.